cnd/src/main.rs

```diff
@@ -52,6 +53,12 @@ fn main() -> anyhow::Result<()> {
 
     crate::trace::init_tracing(settings.logging.level)?;
 
+    let path = &settings.data.dir.to_path_buf();
+
+    let file = std::fs::File::open(&path)?;
+    file.try_lock_exclusive()?;
+    file.lock_exclusive()?;
```

Comment:
Unless I'm missing something, this is not correct.  You should not be locking the file twice.  Just using `try_lock_exclusive()?`.  Also, can we have this all in a function please, so in main it just reads something like `lock_data_dir(&settings.data.dir)?`.  Thanks

Reply:
managed to move into a lock_datadir function. It was wasn't working earlier because you need to keep the file object in scope to maintain the lock

Reply:
also was able to remove the redundant lock statement succesfully